cmake/partition_manager.cmake

```diff
@@ -415,23 +415,26 @@ else()
     ${global_hex_depends}
     )
 
-  # For convenience, generate global hex file containing all domains' hex files.
-  set(final_merged ${PROJECT_BINARY_DIR}/merged_domains.hex)
-
-  # Add command to merge files.
-  add_custom_command(
-    OUTPUT ${final_merged}
-    COMMAND
-    ${PYTHON_EXECUTABLE}
-    ${ZEPHYR_BASE}/scripts/mergehex.py
-    -o ${final_merged}
-    ${domain_hex_files}
-    DEPENDS
-    ${global_hex_depends}
-    )
-
-  # Wrapper target for the merge command.
-  add_custom_target(merged_domains_hex ALL DEPENDS ${final_merged})
+  if (PM_DOMAINS)
+    # For convenience, generate global hex file containing all domains' hex
+    # files.
+    set(final_merged ${PROJECT_BINARY_DIR}/merged_domains.hex)
```

Comment:
Would `ZEPHYR_BINARY_DIR` be a better choice ?

Reply:
Aren't those equivalent?

ref 
```
# zephyr/cmake/app/boilerplate.cmake:97

set(ZEPHYR_BINARY_DIR ${PROJECT_BINARY_DIR})
```

Reply:
I see that in this file `${PROJECT_BINARY_DIR}` is used most, is the convention that we should be using `${ZEPHYR_BINARY_DIR}`?

Reply:
Changed to `ZEPHYR_BINARY_DIR`

Reply:
`ZEPHYR_BINARY_DIR` is fixed, `PROJECT_BINARY_DIR` may potentially change if another `project()` call is made.